### ISSUE TITLE
Region to mask image

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#650]
@OlgaVorokh - Please add a method to SkyImage that takes a region object and fills it's data with 1 for pixels inside the region and 0 outside. Later we can add a `subpixels` option (like there already is in `photutils` e.g. [here](http://photutils.readthedocs.io/en/latest/api/photutils.aperture.get_cutouts.html#photutils.aperture.get_cutouts)) to fill values between 0 and 1 for pixels that are partly inside the region.

My suggestion would be:
```python
>>> image = SkyImage.empty(...)
>>> region = ...
>>> image.fill_region(region)
```
i.e. that `fill_region` modifies `self.data` (and not return a copy).
@adonath @OlgaVorokh - Does this name and behaviour sound good to you or would you prefer something else?

The [ExclusionMask.from_ds9](https://gammapy.readthedocs.io/en/latest/api/gammapy.image.ExclusionMask.html#gammapy.image.ExclusionMask.from_ds9) method should be removed ... `pyregion` will be deprecated and we should just use `region` and `image` objects everywhere.

FYI: My preference is that we get rid of `ExclusionMask` eventually, and just merge the few mask-related methods onto `SkyImage` and add an `is_mask` attribute that checks if `data` is book or numbers with only 0, 1. But we can discuss this later, for now the argument to have `fill_region` on `SkyImage` is that this allows to have a `subpixel` option later with partial pixel coverage floats.

**Comment by cdeil** [comment]
Concerning implementation I'm not sure yet  ... there's two options:
1. check if the input region is a pixel or sky region. if it's a pixel region, create pixel coordinate grids and call `region.contains` on those. if it's a sky region, create sky coordinate grids and call `region.contains` on those.
2. always transform the region to pix coord on input
3. always transform the region to sky coord on input

I think my preference is 1., but I'm not sure.
@adonath - Any thoughts on this question?

**Comment by adonath** [comment]
I think adding `fill_region()` to `SkyImage` only makes sense, when the `ExclusionMask` is finally (definitely) removed in a subsequent later PR. The way the code is structured right now, `fill_region()` belongs to the `ExclusionMask` class. So my preference would be to add the `fill_region()` method there and copy over the binary functions such as `dilate`, `erode`, `open` and `close` from `image.utils` as well. 

Whether we should get rid of the `ExclusionMask` class is separate discussion. When we decide to merge it with `SkyImage`, we can just copy it over and resolve `ExclusionMask`, in a single new PR. 
Maybe let's iclude @joleroi in the discussion, because he used `ExclusionMask` and regions the most. 

Concerning the second question: I prefer option 1 as well.

**Comment by joleroi** [comment]
The idea behind the ``from_ds9`` method was that users might have a list of exclusion regions rather than a boolean mask. Of course the implementation should change, so that a regions list is created from the ds9 files using the functionality in ``regions`` and then your proposed ``fill_region`` is called. I don't think it makes much sense to have this method on the ``SkyImage`` class, since it is very specific to the excluded regions use case. Also the ``distance_image`` function is somehow special to the excluded regions case.

So my preference would be to leave the ``ExclusionMask``, unless you have a good solution for the aforementioned examples (and other stuff that's really specific to exclusion masks).

Concerning the second question: I guess my vote doesn't counts anymore, but I don't care anyway so that's fine :smile:

**Comment by cdeil** [comment]
OK, let's put this on `ExclusionMask` for now if @adonath and @joleroi prefer.

@OlgaVorokh - to implement it, please do this:
- use `.data` and not the `.mask` property from `ExclusionMask`. The `mask` property is a bad idea anyways and should be removed, again a discussion for a future PR.
- fill 1 for inside and 0 for outside. Leave the discussion of whether to switch that meaning of data values for `ExclusionMask` to a future PR.

**Comment by joleroi** [comment]
What do you think about renaming ``ExclusionMask`` to ``SkyImageMask`` or ``SkyMask``? Then this is really only a special case for a ``SkyImage`` with additional functionality.

**Comment by cdeil** [comment]
+3 to remove `ExclusionMask` completely.

If that doesn't find consensus, then yes, `+1` to rename to `SkyMask` and remove the `mask` property as an alias to `data`
@adonath - Is that OK with you?

**Comment by cdeil** [comment]
@OlgaVorokh - After #652 the remaining task here is to add `SkyMask.fill_region` and a test as described above.

**Comment by cdeil** [comment]
Done in #655  and https://github.com/gammapy/gammapy/commit/f728810c468508eafba22a205a49ba2523366692

See https://github.com/gammapy/gammapy/pull/655#issuecomment-236374259 for my thoughts on the API that's now in master.
